src/cargo/util/errors.rs

```diff
@@ -286,6 +286,20 @@ impl From<Box<CargoError>> for CliError {
 }
 
 // =============================================================================
+// NetworkError trait
+
+pub trait NetworkError: CargoError {
+    fn maybe_spurious(&self) -> bool;
+}
+
+impl NetworkError for git2::Error {
+    fn maybe_spurious(&self) -> bool { true }
```

Comment:
For git errors this may want to at least check the [`class`](http://alexcrichton.com/git2-rs/git2/struct.Error.html#method.class) of the error for `Net`, and perhaps look at the `code` as well to ensure it's not a known non-spurious error.